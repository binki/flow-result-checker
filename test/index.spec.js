const { spawnSync } = require('child_process')
const { resolve } = require('path')
const createChecker = require('../checker')

describe('main tool', () => {
  it('should display help', () => {
    const { stdout } = spawnSync('node', ['index.js', '--help'], { encoding: 'utf8' })

    expect(stdout.replace(/ +/g, ' ')).toMatchSnapshot();
  })

  it('should find no errors in `index.js`', () => {
    const expected = `Found 0 errors
`

    const { stdout } = spawnSync('flow', ['check'], { encoding: 'utf8' })

    expect(stdout.replace(/ +/g, ' ')).toEqual(expected.replace(/ +/g, ' '))
  })
})
