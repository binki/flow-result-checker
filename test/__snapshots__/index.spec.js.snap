exports[`main tool should display help 1`] = `
"
Usage:index[options]


Options:

-V,--versionoutputtheversionnumber
--no-color
--debugging
-d,--dir<dir>Thedirectorywheretosavetheflowresultanderrors.Ifnotspecifiednofileswillbesaved.
-f,--file<filename>Nameoftheresultfile.defaultsto\'results.txt\'
-e,--error-file<filename>Nameofthefilewhereparsingerrorsarestored.defaultsto\'error.txt\'
-h,--helpoutputusageinformation
"
`;
